lua/starfall/libs_sh/game.lua

```diff
@@ -130,10 +142,88 @@ if SERVER then
 	-- @param number damageRadius The radius in which entities will be damaged (0 - 1500)
 	-- @param number damage The amount of damage to be applied
 	function game_library.blastDamage(damageOrigin, damageRadius, damage)
-		checkpermission(instance, nil, "blast.create")
+		checkpermission(instance, nil, "game.blastDamage")
 		util.BlastDamage(instance.entity, instance.player, vunwrap(damageOrigin), math.Clamp(damageRadius, 0, 1500), damage)
 	end
 
+	--- Fires a bullet. Bullet made with this function will not have any tracer, you will have to make them yourself.
+	-- @server
+	-- @param Vector src The position to fire the bullets from.
+	-- @param Vector Dir The fire direction.
+	-- @param number? damage The damage dealt by the bullet. Default: (1-100)
+	-- @param number? num The amount of bullets to fire. Default: (1-40)
+	-- @param number? force The force of the bullets. Default: (0-100)
+	-- @param number? distance Maximum distance the bullet can travel.
+	-- @param Vector? Spread The spread, only x and y are needed.
+	-- @param number? hullSize The hull size of the bullet. Default: (0-10)
+	-- @param Entity? ignoreEntity The entity that the bullet will ignore when it will be shot.
+	-- @param function? callback Function to be called with attacker, traceResult after the bullet was fired but before the damage is applied (the callback is called even if no damage is applied).
+	function game_library.bulletDamage(src, dir, damage, num, force, distance, spread, hullSize, ignoreEntity, cb)
+		checkpermission(instance, nil, "game.bulletDamage")
+		if damage ~= nil then checkluatype(damage, TYPE_NUMBER) damage = math.Clamp(damage, 1, maxBulletDamage:GetInt()) else damage = 1 end
+		if force ~= nil then checkluatype(force, TYPE_NUMBER) force = math.Clamp(force, 0, maxBulletForce:GetInt()) else force = 0 end
+		if distance ~= nil then checkluatype(distance, TYPE_NUMBER) distance = math.Clamp(distance, 0, 32768) else distance = 32768 end
+		if hullSize ~= nil then checkluatype(hullSize, TYPE_NUMBER) hullSize = math.Clamp(hullSize, 0, maxBulletHull:GetInt()) else hullSize = 0 end
+		if num ~= nil then checkluatype(num, TYPE_NUMBER) num = math.Clamp(num, 1, maxBulletNum:GetInt()) else num = 1 end
```

Comment:
```suggestion
		if damage ~= nil then checkluatype(damage, TYPE_NUMBER) damage = math.Clamp(damage, 0, fireBulletsDPSBurst.max) else damage = 1 end
		if force ~= nil then checkluatype(force, TYPE_NUMBER) force = math.Clamp(force, 0, maxBulletForce:GetInt()) else force = 0 end
		if distance ~= nil then checkluatype(distance, TYPE_NUMBER) distance = math.Clamp(distance, 0, 32768) else distance = 32768 end
		if hullSize ~= nil then checkluatype(hullSize, TYPE_NUMBER) hullSize = math.Clamp(hullSize, 0, maxBulletHull:GetInt()) else hullSize = 0 end
		if num ~= nil then checkluatype(num, TYPE_NUMBER) num = math.Clamp(num, 1, fireBulletsBurst.max) else num = 1 end
```

Can get rid of the damage and numbullet cvars since the burst objects maxes work for those

Reply:
Are you sure damage min clamp should be 0 ? Since if a bullet damage is set to 0 it will use the ammotype to calculate the damage of the bullet, in our case the ammo type will always be nil be I'm not sure how it handle it. And any damage calculated this way will not count towards the dps burstobject.

Reply:
I wasn't aware 0 would make it use the default ammotype's damage; that's kind of lame. Nevermind then

Reply:
I'll test now

Reply:
If you can get rid of those last two cvars, then it should be ready to merge.